app/src/main/kotlin/com/wire/android/ui/home/HomeScreen.kt

```diff
@@ -21,77 +21,94 @@ import androidx.compose.runtime.rememberCoroutineScope
-                )
-                // We are not including the topBar in the Scaffold to correctly handle the collapse scroll effect on the search,
-                // which will not be possible when using Scaffold topBar argument
-                topBar()
+        NavigationDrawer(
+            drawerContainerColor = MaterialTheme.colorScheme.surface,
+            drawerTonalElevation = 0.dp,
+            drawerShape = RectangleShape,
+            drawerState = drawerState,
+            drawerContent = drawerContent,
+            gesturesEnabled = currentNavigationItem.isSwipeable
+        ) {
+
+            val homeContent: @Composable () -> Unit = {
+                Box {
+                    val startDestination = HomeNavigationItem.all.firstOrNull { startScreen == it.route }?.route
+
+                    HomeNavigationGraph(
+                        homeState = homeState,
+                        navController = navController,
+                        startDestination = startDestination
+                    )
+
+                    HomeTopBar(
+                        title = currentNavigationItem.title,
+                        isSearchable = currentNavigationItem.isSearchable,
+                        scrollPosition = scrollPosition,
+                        onUserProfileClick = { viewModel.navigateToUserProfile() },
+                        onHamburgerMenuItemCLick = { openDrawer() }
+                    )
+                }
             }
-        }
 
-        val homeScreen: @Composable () -> Unit = homeState.homeBottomSheetContent?.run {
-            {
+            val homeBottomSheetContent = homeBottomSheetContent
+
+            if (homeBottomSheetContent != null) {
                 WireModalSheetLayout(
-                    sheetState = homeState.bottomSheetState,
-                    sheetContent = this
+                    sheetState = bottomSheetState,
+                    sheetContent = homeBottomSheetContent
                 ) {
                     homeContent()
                 }
+            } else {
+                homeContent()
             }
-        } ?: { homeContent() }
 
-        homeScreen()
+        }
+
+        BackHandler(enabled = drawerState.isOpen) { closeDrawer() }
     }
-    BackHandler(enabled = drawerState.isOpen) { scope.launch { drawerState.close() } }
 }
 
-
-@OptIn(ExperimentalMaterialApi::class)
+@OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class)
 class HomeState(
-    private val coroutineScope: CoroutineScope,
+    val coroutineScope: CoroutineScope,
+    val navController: NavHostController,
+    val drawerState: DrawerState,
     val bottomSheetState: ModalBottomSheetState,
-    bottomSheetContent: @Composable (ColumnScope.() -> Unit)?
+    bottomSheetContent: @Composable (ColumnScope.() -> Unit)?,
+    val currentNavigationItem: HomeNavigationItem
 ) {
 
+    var scrollPosition by mutableStateOf(0)
+        private set
```

Comment:
I set the setter to be private so that we update the scroll position with more "readable", "clean" method : updateScrollPosition(newScrollPosition: Int) :P just a personal touch in my opinion ;p